catalyst/callbacks/profiler.py

```diff
@@ -0,0 +1,228 @@
+        export_chrome_trace_path: str = None,
+        export_stacks_kwargs: Dict[str, Any] = None,
+    ):
+        super().__init__(order=CallbackOrder.Internal, node=CallbackNode.Master)
+
+        self.loader_key = loader_key
+        self.epoch = epoch
+        self.num_batches = num_batches
+        self.batch_cnt = 0
+
+        self.profiler_kwargs = {} if profiler_kwargs is None else profiler_kwargs
+        if tensorboard_path is not None and "on_trace_ready" not in profiler_kwargs:
+            self.profiler_kwargs["on_trace_ready"] = torch.profiler.tensorboard_trace_handler(
+                tensorboard_path
+            )
+        self.export_chrome_trace_path = export_chrome_trace_path
+        self.export_stacks_kwargs = export_stacks_kwargs
+        self.profiler = None
+        self.stats = None
+
+    def on_experiment_start(self, runner: IRunner) -> None:
+        """
+        On batch end action
+
+        Args:
+            runner: current runner
+        """
+        if self.loader_key is None:
+            self.loader_key = runner.loader_key  # use first loader for profile
+
+    def _should_use_profiler(self, loader_key: str, epoch: int):
+        if self.loader_key == loader_key and self.epoch == epoch:
+            if self.num_batches is not None:
+                return self.batch_cnt < self.num_batches
+            return True
+        return False
+
+    def _enter_profiler(self, loader_key: str, epoch: int) -> None:
+        if not self._should_use_profiler(loader_key, epoch):
+            return
+
+        if self.profiler is None:
+            self.profiler = torch.profiler.profile(**self.profiler_kwargs)
+            self.profiler.__enter__()
+
+    def _exit_profiler(
+        self, loader_key: str, epoch: int, loggers: Sequence[ILogger] = None
+    ) -> None:
+        if not self._should_use_profiler(loader_key, epoch) or self.profiler is None:
+            return
+
+        if self.stats is None:
+            loggers_ = [] if loggers is None else loggers  # noqa: WPS120
+
+            self.profiler.__exit__(None, None, None)
+
+            if "on_trace_ready" not in self.profiler_kwargs and self.export_chrome_trace_path:
+                self.profiler.export_chrome_trace(self.export_chrome_trace_path)
+
+            if self.export_stacks_kwargs is not None:
+                self.profiler.export_stacks(**self.export_stacks_kwargs)
+
+            self.stats = self.profiler.key_averages()
+            table_txt = self.stats.table(sort_by="cpu_time_total", row_limit=100)
+
+            with TemporaryDirectory() as tmp_dir:
+                artifact_path = os.path.join(tmp_dir, "profiler_table.txt")
+                with open(artifact_path, "w") as f:
+                    f.write(table_txt)
+                for logger in loggers_:
+                    logger.log_artifact(
+                        tag="profiler", artifact="profiler.txt", path_to_artifact=artifact_path,
+                    )
```

Comment:
maybe, let's use `runner.log_artifact` instead?

Reply:
Yeah, I'll change that.
Btw, will it work correctly when the user somehow doesn't specified loggers (i.e. dict with loggers is empty)?

Reply:
Yeah, I'll change that.
Btw, will it work correctly when the user somehow doesn't specified loggers (i.e. dict with loggers is empty)?

Reply:
yup :)